eng/tools/typespec-validation/src/rules/sdk-tspconfig-validation.ts

```diff
@@ -427,12 +462,23 @@ export class TspConfigCsharpMgmtPackageDirectorySubRule extends TspconfigEmitter
   }
 }
 
+export class TspConfigCsharpMgmtServiceDirMatchPatternSubRule extends TspconfigEmitterOptionsSubRuleBase {
+  constructor() {
+    super("@azure-tools/typespec-csharp", "service-dir", new RegExp(/^sdk\/.*$/));
```

Comment:
the missing rule for .net is `namespace === package-dir`, you should check whether the `namespace` option's value is exactly the same to `package-dir` option